web/src/components/Button.tsx

```diff
@@ -1,54 +1,139 @@
 import React from 'react';
 import { twMerge } from 'tailwind-merge';
 
-// This value is selected based on appropriate size given most labels in most languages.
-const BUTTON_MIN_WIDTH_CLASS = 'min-w-[232px]';
-
-// TODO: This should extend either HTMLButtonElement or HTMLAnchorElement, but I could not get it
-// to properly select the correct one based on given props.
 interface ButtonProps {
   icon?: React.ReactNode;
   children?: React.ReactNode;
   disabled?: boolean;
-  background?: string;
-  textColor?: string;
+  size?: 'sm' | 'lg' | 'xl';
+  type?: 'primary' | 'secondary' | 'tertiary' | 'link';
   href?: string;
-  className?: string;
+  backgroundClasses?: string;
+  foregroundClasses?: string;
   onClick?: () => void;
 }
 
 export function Button({
   icon,
   children,
   disabled,
-  background,
-  textColor,
   href,
-  className,
+  backgroundClasses, // backgroundColor, borderColor, margin, etc.
+  foregroundClasses, // textColor, etc.
+  size = 'lg',
+  type = 'primary',
   onClick,
 }: ButtonProps) {
   const renderAsLink = Boolean(href);
   const As = renderAsLink ? 'a' : 'button';
-  const type = renderAsLink ? undefined : 'button';
+  const componentType = renderAsLink ? undefined : 'button';
   const isIconOnly = !children && Boolean(icon);
 
   return (
-    <As
+    <div
       className={twMerge(
-        `mx-2 my-3 flex min-h-[45px] w-fit items-center justify-center gap-x-2 rounded-full bg-white px-2 py-3 text-md font-bold shadow-[0px_0px_13px_rgb(0_0_0/12%)] transition duration-200 hover:shadow-[0px_0px_23px_rgb(0_0_0/20%)] dark:bg-gray-600 dark:hover:shadow-[0px_0px_23px_rgb(0_0_0/50%)]`,
-        !isIconOnly && BUTTON_MIN_WIDTH_CLASS,
-        `${disabled ? 'opacity-60 hover:shadow-[0px_0px_13px_rgb(0_0_0/12%)]' : ''}`,
-        className
+        `items-center justify-center rounded-full ${getBackground(type, disabled)}`,
+        backgroundClasses
       )}
-      translate="no"
-      disabled={disabled}
-      style={{ color: textColor, background: background }}
-      href={href}
-      type={type}
-      onClick={onClick}
     >
-      {icon}
-      {children}
-    </As>
+      <As
+        className={twMerge(
+          `flex h-full w-full flex-row items-center justify-center rounded-full text-sm font-semibold focus:bg-inherit
```

Comment:
I'm not sure if we want the `focus:bg-inherit`. I added it because I thought it looked better on mobile, when clicking the 'hide wind/solar layer' and 'color blind mode', so they do not look hovered when they have been pressed. But on web the hover disappear when the button has been clicked, and idk if we want that. What do you think is the best solution @madsnedergaard @VIKTORVAV99 ?

Reply:
I think that's okay for now since we'll be moving to the switches later on anyway :)

Reply:
I agree with Mads, let's not let such small details hold us back when we know we will be changing it soon anyhow.